src/app/plugins/help.plugin.ts

```diff
@@ -23,54 +23,55 @@ export class HelpPlugin extends Plugin {
     const commands = this.container.pluginService.aliases;
     const input: string = this._parseCommand(args || []);
 
-    if (!!commands[input]) {
+    if (commands[input]) {
       const pluginName = commands[input];
       const plugin = this.container.pluginService.plugins[pluginName];
 
       if (plugin.permission === ChannelType.Admin || plugin.permission === ChannelType.Staff) {
-        if (this._embed[1].fields!.length === 0) {
-          this._generateEmbed(false, 1);
+        // as the admin/mod commands aren't meant to be known this just shows the basic help as if nothing happened.
+        if (this._embed[0].fields!.length === 0) {
+          this._generateEmbed(0);
         }
 
-        message.reply(this._embed[1]);
+        message.reply(this._embed[0]);
       } else {
         if (this._pluginEmbed[pluginName] === undefined) {
           this._pluginEmbed[pluginName] = new RichEmbed();
           this._generatePluginEmbed(pluginName);
         }
         message.reply(this._pluginEmbed[pluginName]);
       }
-    } else if (input === `all`) {
-      if (this._embed[0].fields!.length === 0) {
-        this._generateEmbed(true, 0);
+    } else if (input === 'all') {
+      if (this._embed[1].fields!.length === 0) {
+        this._generateEmbed(1);
       }
 
-      message.reply(this._embed[0]);
+      message.reply(this._embed[1]);
     } else {
-      if (this._embed[1].fields!.length === 0) {
-        this._generateEmbed(false, 1);
+      if (this._embed[0].fields!.length === 0) {
+        this._generateEmbed(0);
       }
 
-      message.reply(this._embed[1]);
+      message.reply(this._embed[0]);
     }
   }
 
-  private _generateEmbed(all: boolean, targ: number) {
+  private _generateEmbed(targ: number) {
```

Comment:
You just use `targ` as 0 or 1? it should be boolean then, and have a more meaningful name